website/app/modifiers/doc-track-event-on.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Modifier from 'ember-modifier';
+import { registerDestructor } from '@ember/destroyable';
+import { inject as service } from '@ember/service';
+
+function handleTriggerEvent(eventName) {
+  // https://usefathom.com/docs/features/events
+  window.fathom?.trackEvent(eventName);
+}
+
+function cleanup(instance) {
+  instance.element.removeEventListener(
+    instance.triggerEvent,
+    handleTriggerEvent
+  );
+
+  instance.element = null;
+  instance.eventName = null;
+  instance.triggerEvent = null;
+}
+
+export default class DocTrackEventOn extends Modifier {
```

Comment:
My understanding is that we can't use a [function-based modifier](https://github.com/ember-modifier/ember-modifier?tab=readme-ov-file#function-based-modifiers) here because we're using the `@service`, is that right?

Reply:
Yes, exactly